### PR TITLE
Nobreath trait now properly allows things to heal oxyloss

### LINKED ISSUE #76804: Revived androids can be stuck with un-healable oxy damage
## Testmerges: 

#76728 

## Reproduction:

Find a corpse, convert it into an android with the robotic chaplain sect, then attempt to defibrilate. The android will be revived, but will have 150 oxy damage that does not heal on its own and seemingly cannot be healed, at least not with the methods tested (bible smacking and CPR). As far as I know, androids do not breath and should not be capable of getting oxy damage in the first place.

I've converted and then revived corpses before, so this bug was likely caused by a recent change. Probably one of the species/organ refactors.

### PR DESCRIPTION
**Pull request by GPeckman**
## About The Pull Request

Previously, the nobreath trait would constantly clear oxyloss and suffocation damage from people with it. However, this functionality was removed apparently by accident in #76209. Rather than re-adding that functionality, this PR causes creatures with nobreath to regenerate oxyloss at a fixed rate, as there are sources of oxyloss other than suffocation (bloodloss). This also fixes #76804
## Why It's Good For The Game

Bugs bad.
## Changelog
:cl:
fix: Creatures with nobreath trait will now regenerate oxyloss. Androids, skeletons and such will no longer be stuck with un-healable damage after being revived.
/:cl:

### CODE REVIEW

**Comment by GPeckman** [comment]
Under previous behavior, nobreath granted full immunity to oxyloss. The comment documenting it says the same: https://github.com/tgstation/tgstation/blob/2baf9dbf5720b8ffffc1d7961153491dd3a2630e/code/__DEFINES/traits.dm#L277-L278
There's an argument to be made that this should be changed, but that might be better done in a different PR

**Review by MrMelbert — CHANGES_REQUESTED** [review]
I think this is a bad fix, `on_nobreath_trait_gain` sets oxyloss to 0 and losebreath when you gain the trait so nothing should be carried over, that means clearly there are some cases of applyin oxyloss that does not account for nobreath - which should be addressed instead

**Comment by GPeckman** [comment]
I've tested the bug (the unhealable oxyloss one) in more detail on a local server, and its pretty cursed. So, the unhealable oxyloss appears to happen if two conditions are met: A carbon must have TRAIT_NOBLOOD and it must have TRAIT_NOBREATH with species as the source. Having a different source of the trait causes a slightly different variation of the bug: oxyloss will heal, but the 'health' of a mob (seen in the heart monitor UI indicator on the right, and in the overall status line when using a health analyzer) will not change. If a carbon has TRAIT_NOBREATH from both species and a non-species source, then this new variant of the bug will still occur.

It's probably going to take a while to track down the exact cause of this.

**Comment by GPeckman** [comment]
Finally figured out the bug, it wasn't related to TRAIT_NOBLOOD at all. It was a false positive due to the way I tested things.

I figured out the _real_ cause: the code causing humans to breathe was skipped entirely if the human had TRAIT_NOBREATH. Now, that code will still be skipped but the human in question will regenerate oxyloss. The rate of regeneration is equal to what a normal human could do under ideal circumstances (5 oxyloss per breathing cycle).

**Comment by ATH1909** [comment]
I still think that setting their oxyloss to 0 each tick (and/or making them immune to taking oxyloss damage) is better, as it seems kind of weird to me that, say, chugging can cause a creature that doesn't need to breathe to suffocate.

**Comment by nikothedude** [comment]
> I still think that setting their oxyloss to 0 each tick (and/or making them immune to taking oxyloss damage) is better, as it seems kind of weird to me that, say, chugging can cause a creature that doesn't need to breathe to suffocate.

But what about creatures with blood? This would make things like silverscales imprevious to blood loss or literally ANY OTHER SOURCE of oxyloss.

**Comment by MrMelbert** [comment]
> I still think that setting their oxyloss to 0 each tick (and/or making them immune to taking oxyloss damage) is better, as it seems kind of weird to me that, say, chugging can cause a creature that doesn't need to breathe to suffocate.

Setting oxyloss = 0 every tick is just wasting processing on nothing, nobeath should just prevent oxygen damage from accumulating period

**Comment by GPeckman** [comment]
I've changed the documentation comment for nobreath, now it specifies that it does not grant full immunity to oxy damage.

With that said, would it make sense to add a new trait that does grant full immunity like that? There's already one for toxin damage.

**Comment by distributivgesetz** [comment]
This fixes #77127

**Comment by GPeckman** [comment]
It seems I misunderstood what Melbert wanted. TRAIT_NOBREATH no _does_ grant full oxyloss immunity, and does it by adding a check to the adjustOxyLoss proc instead of clearing oxyloss every tick.
